hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/container/replication/ReplicationManagerUtil.java

```diff
@@ -178,6 +188,17 @@ An UNHEALTHY replica with unique origin node id of a QUASI_CLOSED container shou
           excludedNodes.add(r.getDatanodeDetails());
           continue;
         }
+        if (nodeStatus.isDead()) {
```

Comment:
Here should we add `nodeStatus.isMaintenance() && nodeStatus.isDead()`? It is probably assumed that only a maintenance node can be dead and have still a replica in the system, but just incase that changes we should be explicit about it here.

Reply:
Done, also added some basic unit testing.